river/Cursor.zig

```diff
@@ -768,22 +780,33 @@ fn processMotion(self: *Self, device: *wlr.InputDevice, time: u32, delta_x: f64,
                 self.wlr_cursor.y - @intToFloat(f64, output_box.y + view.current.box.y - view.surface_box.y),
             );
         },
-        .move => |view| {
-            view.move(@floatToInt(i32, delta_x), @floatToInt(i32, delta_y));
+        .move => |*data| {
+            dx += data.delta_x;
+            dy += data.delta_y;
+            data.delta_x = dx - @round(dx);
+            data.delta_y = dy - @round(dy);
```

Comment:
We need to use trunc here instead since round can round up.
For instance with using round, if our "new" dx value were 0.9, it would round to 1.0 which would mean data's delta_x is going to be -0.1. If dx was 0.9 again, our "new" dx value would be 0.8 when it should be 1.8 (0.9 + 0.9 = 1.8 just in case my explanation wasn't clear).

```suggestion
            data.delta_x = dx - @trunc(dx);
            data.delta_y = dy - @trunc(dy);
```

Reply:
Great catch, the semantics of `@trunc()` are indeed what I intended here!